blog/_posts/2018-05-02-developing-on-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+* As of Draft version 0.12.0, builds are executed locally
+* User doesn’t have an option to choose something other than Helm for deployment
+* It can watch local changes and trigger deployments, but this is feature is not enabled by default
+* It allows developer to use either local or remote Kubernetes cluster
+* Deploying to production is up to the user, Draft authors recommend their other project – Brigade
+* Can be used instead of Skaffold, and along the side of Squash
+
+More info:
+[Draft: Kubernetes container development made easy](http://blog.kubernetes.io/2017/05/draft-kubernetes-container-development.html)
+[Getting Started Guide](https://github.com/Azure/draft/blob/master/docs/getting-started.md)
+
+### Skaffold
+
+[Skaffold](https://github.com/GoogleCloudPlatform/skaffold) is a tool that aims to provide portability for CI integrations with different build system, image registry and deployment tools. It is different from Draft, yet somewhat comparable. It has a basic capability for generating manifests, but it’s not a prominent feature. Skaffold is extendible and lets user pick tools for use in each of the steps in building and deploying their app.
+
+Implications:
+* Modular and design
+* Works independently of CI vendor, user doesn’t need Docker or Kubernetes plugin
+* Works without CI as such, i.e. from the developer’s laptop
+* It can watch local changes and trigger deployments
+* It allows developer to use either local or remote Kubernetes cluster
+* It can be used to deploy to production, user can configure how exactly they prefer to do it and provide different kind of pipeline for each target environment
+* Can be used instead of Draft, and along the side with most other tools
+
+More info:
+* [Introducing Skaffold: Easy and repeatable Kubernetes development](https://cloudplatform.googleblog.com/2018/03/introducing-Skaffold-Easy-and-repeatable-Kubernetes-development.html)
+* [Getting Started Guide](https://github.com/GoogleCloudPlatform/skaffold#getting-started-with-local-tooling)
+
+### Squash
+[Squash](https://github.com/solo-io/squash) consists of a debug server that is fully integrated with Kubernetes, and a IDE plugin. It allows you to insert breakpoints and do all the fun stuff you are used to doing when debugging an application using an IDE. It bridges IDE debugging experience with your Kubernetes cluster by allowing you to attach the debugger to a pod running in your Kubernetes cluster.
+
+Implications:
+* Can be use independently of other tools you chose
+* Requires a privileged DaemonSet
+* Integrates with popular IDEs
+* Supports Go, Python, Node.js, Java and gdb
+* User must ensure application binaries inside the container image are compiled with debug symbols
+* Can be used in combination with any other tools described here
+* It can be used with either local or remote Kubernetes cluster
+
+More info:
+* Squash: A Debugger for Kubernetes Apps
+* Getting Started Guide
```

Comment:
the above two links are missing

Reply:
Squash: A Debugger for Kubernetes Apps -> https://www.youtube.com/watch?v=5TrV3qzXlgI

Getting Started Guide -> https://github.com/solo-io/squash/blob/master/docs/getting-started.md

Reply:
```
* [Squash: A Debugger for Kubernetes Apps](https://www.youtube.com/watch?v=5TrV3qzXlgI)
* [Getting Started Guide](https://github.com/solo-io/squash/blob/master/docs/getting-started.md)
```

Reply:
Fixed